flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamConfig.java

```diff
@@ -735,4 +761,10 @@ public int hashCode() {
 			return inputEdge.hashCode();
 		}
 	}
+
```

Comment:
Why not putting it as a method in the `InputConfig` with a default implementation that returns `false`? I think that this will make the code nicer and also it will avoid checks like `inputConfig instanceof StreamConfig.NetworkInputConfig`

Reply:
I decided to put it into the `NetworkInputConfig` because that's the only place where it can really be applied right now. But I can also move it to the base class.